docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+                        path_recorded = path_main_file
+                        lines = f"\n\t {dirname}/{dir_main}"
+                        outlines += lines
+            fh.write(outlines)
+    # Opens each DirectoryName_main.rst and add the name of the scripts in that directory
+    with open(path_main_file, "a") as fh:
+        fname = os.path.basename(fpath)
+        # extract the "kmers" of kmers.py
+        basename = os.path.splitext(fname)[0]
+        fh.write(f"\n\t {basename}")
+
+
+def link_write_usage():
+    """
+    Creates usage.rst and writes the directories in its toctree
+    """
+    if not os.path.exists(path_usage):
+        with open(path_usage, "w") as fh:
+            write_text = "\n".join(["="*len("usage"), "Usage", "="*len("usage"), " ", ".. toctree::",
+                                    "\t :maxdepth: 3", "\t :caption: Table of Contents", " "])
+            fh.write(write_text)
+            # This can also be done using next() -> https://stackoverflow.com/a/142535/12671809 ,
+            # but reduces readability. What to do ??
+            for dirname in os.listdir(path_scripts):
+                path_dirname = os.path.join(path_scripts, dirname)
+                if os.path.isdir(path_dirname):
+                    dirname_rst = dirname + "_main"
+                    fh.write(f"\n\t {dirname}/{dirname_rst}")
+
+
+def remove_existing_docs():
+    """
+    Removes the exists documentatio in `scripts` directory
+    """
+    if os.path.exists(path_scripts):
+        shutil.rmtree(path_scripts)
+
+
+def remove_empty_dir():
+    """
+    Removes any empty directory that may have been added by mistake
+    """
+    for root, dirnames, filenames in os.walk(os.path.join(os.path.dirname(os.path.dirname(__file__)), "autometa")):
+        if os.listdir(root) == []:
+            os.rmdir(root)
+
+
+def remove_temp_files(tmp_fpath):
+    """
+    Removes the temporary files to which the argparse block was copied
+
+    Parameters
+    ----------
+    tmp_fpath : string
+        path to temporary file that has the argparse block
+        e.g. '</path/to/tmp/file>'
+    """
+    os.remove(tmp_fpath)
+
+
+path_scripts = os.path.join(os.path.dirname(__file__), "source", "scripts")
+path_usage = os.path.join(path_scripts, "usage.rst")
+
+remove_existing_docs()
+remove_empty_dir()
+
+for fpath in glob.glob(os.path.join(os.path.dirname(os.path.dirname(__file__)), "autometa", "**", "*.py"), recursive=True):
+    exclude_dirs = ["validation"]
+    if "__" in fpath:  # do not include __init__ and __main__.py files
+        continue
+    if os.path.basename(os.path.dirname(fpath)) in exclude_dirs:
+        continue
+    tmp_fpath = get_argparse_block(fpath)
+    rst_dirpath = path_dir_docs_scripts(fpath)
+    write_run_argparse_output(tmp_fpath, rst_dirpath, fpath)
+    design_main_file(fpath, rst_dirpath)
+
+link_write_usage()
+remove_temp_files(tmp_fpath)
+
+
+def main(args):
+    pass
```

Comment:
Is this necessary? What is this doing here?

It appears, the functionality you have listed just above this could be placed in this function when calling this script from the command line.